### DISCUSSION TITLE
[Feature] Improve zoom image quality

### DISCUSSION THREAD

**Discussion by pec0ra** [immich-app/immich#3718] [Feature Request]
### The feature

Load the full quality image when zooming in.

The thumbnails are great to scroll quickly through the pictures, but lack details when zooming. To enjoy the full potential of high resolutions photos, it would be better if the full resolution image was loaded when a user starts to zoom.

An example of this feature is implemented in google photos. If you zoom in on a photo, you will see that after a short time (depending on your connection), a better quality is displayed. 

#### Considerations

Loading the full quality image will of course consume more bandwidth but I think it is safe to assume that when users zoom in on an image, they want the best resolution possible.

Adding an administration setting to disable this feature would be useful for server administrator who don't have the extra bandwidth for this feature.

Note that setting a higher JPEG thumbnail resolution (available since v1.73.0) is not an alternative solution since it would reduce the experience and increase the bandwidth required when scrolling through pictures without zooming.

#### Nice to have

Google photo seem to only load/render the part of the full resolution photo that is displayed on the screen. When moving across the image for the first time at full zoom, you can see that there is a small delay before showing the full resolution. A pixelated version is shown first.

This is of course optimal for speed and bandwidth, but the implementation complexity might not be worth it.

### Platform

- [ ] Server
- [X] Web
- [X] Mobile

**Comment by waclaw66** [comment]
The solution as I observed from Google Photos is to cut the original photo to 512x512 px tiles and load them when the visible area is zoomed enough. Not easy to implement but definitely the most beneficial feature and the reason why I haven't left Google Photos. This also affects server side.

**Comment by waclaw66** [comment]
There has been introduced a solution for web side within https://github.com/immich-app/immich/pull/4040.
